servers/src/main/java/tachyon/master/journal/JournalTool.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.master.journal;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.commons.lang3.StringUtils;
+
+import tachyon.proto.journal.Journal.JournalEntry;
+
+/**
+ * Tool for reading the journal. It reads binary journal entries and prints human-readable ones to
+ * standard out. Example usage below.
+ *
+ * <pre>
+ * java -cp assembly/target/tachyon-assemblies-0.9.0-SNAPSHOT-jar-with-dependencies.jar \
+ *   tachyon.master.journal.JournalTool journal/FileSystemMaster/log.out
+ * </pre>
+ */
+public final class JournalTool {
+  /** Number of dashes to place on a newline at the end of each journal entry. */
+  private static final int NUM_DASHES = 80;
+  /** Amount of time to wait before giving up on the user supplying a journal log via stdin. */
+  private static final long TIMEOUT_MS = 500;
+
+  public static void main(String[] args) throws FileNotFoundException, IOException {
+    InputStream inStream = null;
+    if (args.length == 0 && stdinHasData()) {
+      inStream = System.in;
+    } else if (args.length == 1) {
+      inStream = new FileInputStream(args[0]);
```

Comment:
Good point, that will usually be the case, so I've removed support for specifying the file on the command line. It was complicating the code and it isn't hard to just insert a "<" before the file argument.